src/pages/Entries/ScrollableCalendar.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from "react";
+import { useMemo, useRef } from "react";
+import { useInfiniteQuery } from "react-query";
+import HappinessCard from "./HappinessCard";
+import InfiniteScroll from "react-infinite-scroll-component";
+import { useApi } from "../../contexts/ApiProvider";
+import Spinner from "../../components/Spinner";
+import { Happiness } from "../../data/models/Happiness";
+import { formatDate } from "../../utils";
+
+// type of the paginated happiness data structure used for infinite scroll
+interface HappinessPagination {
+  data: Happiness[];
+  page: number;
+}
```

Comment:
I think this type will be reused later in the application, so it shouldn't just be in the scrollable calendar component file. This kind of stuff is what we are trying to avoid in the rewrite.